packages/auto-form/src/FormBuilder.tsx

```diff
@@ -0,0 +1,185 @@
+            type={'select'}
+            {...props}
+            component={SelectComponent}
+          />
+        );
+      case 'textarea':
+        return (
+          <Field
+            key={props.property.name}
+            name={props.property.name}
+            type={'select'}
+            {...props}
+            component={TextAreaComponent}
+          />
+        );
+      default:
+        return (
+          <pre key={props.property.name}>
+            {props.property.name} = {props.value}
+          </pre>
+        );
+    }
+  };
+  public render() {
+    const initialValues = { ...this.props.initialValues };
+    const fields = [] as IConfigurationProperty[];
+    Object.keys(this.props.definition).forEach(key => {
+      fields.push({ name: key, ...this.props.definition[key] });
+      if (!(key in initialValues)) {
+        initialValues[key] = this.props.definition[key].defaultValue;
+      }
+    });
+    return this.props.children({
+      fields: fields.sort((a, b) => {
+        const aOrder = (a.order || 0) as number;
+        const bOrder = (b.order || 0) as number;
+        return aOrder - bOrder;
+      }),
+      initialValues,
+      onSave: this.props.onSave,
+      renderField: this.renderField,
+    });
+  }
+}
+
+// TODO move these to the UI package
+const InputComponent = ({
+  field,
+  form: { touched, errors },
+  ...props
+}: /*TODO*/ {
+  [name: string]: any;
+}) => (
+  <div className="form-group">
+    <label className="col-sm-2 control-label" htmlFor={field.name}>
+      {props.property.displayName}
+    </label>
+    <div className="col-sm-10">
+      <input data-testid={field.name} {...field} />
+      {touched[field.name] && errors[field.name] && (
+        <div className="error">{errors[field.name]}</div>
+      )}
+    </div>
+  </div>
+);
+
+const SelectComponent = ({
+  field,
+  form: { touched, errors },
+  ...props
+}: /*TODO*/ {
+  [name: string]: any;
+}) => (
+  <div className="form-group">
+    <label className="col-sm-2 control-label" htmlFor={field.name}>
+      {props.property.displayName}
+    </label>
+    <div className="col-sm-10">
+      <select data-testid={field.name} {...field}>
+        {props.property.enum.map((opt: any) => (
+          <option key={opt.value} value={opt.value}>
+            {opt.label}
+          </option>
+        ))}
+      </select>
+      {touched[field.name] && errors[field.name] && (
+        <div className="error">{errors[field.name]}</div>
+      )}
+    </div>
+  </div>
+);
+
+const TextAreaComponent = ({
+  field,
+  form: { touched, errors },
+  ...props
+}: /*TODO*/ {
+  [name: string]: any;
+}) => (
+  <div className="form-group">
+    <label className="col-sm-2 control-label" htmlFor={field.name}>
+      {props.property.displayName}
+    </label>
+    <div className="col-sm-10">
+      <textarea data-testid={field.name} {...field} />
+      {touched[field.name] && errors[field.name] && (
+        <div className="error">{errors[field.name]}</div>
```

Comment:
Once error message has been provided, we should link the error messages to form controls using something like `aria-describedby="errorDesc"` which points to markup looking something like this

`<span id="expDesc">Error: Date is required.</span>`

This will ensure screen reader users hear the error message, otherwise it's just error without explaining what's wrong or how to fix.

Reply:
So I'm gonna defer this to a subsequent issue that I'll raise about moving the form widgets to the UI package and using patternfly react, what's in this PR is an initial effort focused on JSON -> crappy form :-)